packages/components/src/pages/VsCodeExtension.vue

```diff
@@ -312,71 +489,120 @@ export default {
         return callTree;
       }, []);
 
-      if (this.rootCodeObject) {
-        const eventBranches = this.rootCodeObject.allEvents.map((e) => [
-          e.id,
-          e.linkedEvent.id,
-        ]);
+      if (this.filters.rootObjects.length) {
+        this.filters.rootObjects.forEach((id) => {
+          const filterRegExp = new RegExp(id, 'ig');
+          classMap.codeObjects.forEach((codeObject) => {
+            if (!filterRegExp.test(codeObject.fqid)) {
+              return;
+            }
+
+            const eventBranches = codeObject.allEvents.map((e) => [
+              e.id,
+              e.linkedEvent.id,
+            ]);
```

Comment:
Make sure `CodeObject.allEvents` is capturing everything we need. Will a package contain all the events of its ancestors? What about a class that contains nested classes (Such as `app/controllers/Spree` in the demo data). If `allEvents` is empty, it seems like we should skip the filter, because we don't want to end up with no data displayed.

Reply:
Yes, `CodeObject.allEvents` contains all object events and all object children events.
https://github.com/applandinc/appmap-js/blob/main/packages/models/src/codeObject.js#L135-L137